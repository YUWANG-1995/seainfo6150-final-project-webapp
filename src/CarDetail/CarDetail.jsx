import React, { useState, useEffect } from "react";
import { isEmpty } from "lodash";
import { Link } from "react-router-dom";
import styles from "./CarDetail.module.css";

const CarDetail = (props) => {
  const [fetchedCars, setFetchedCars] = useState({});

  useEffect( () => {
    const fetchCars = async () => {
      const result = await fetch("http://demo1468395.mockable.io/brands")
      const response = await result.json();
      setFetchedCars(response);
    };

    if(isEmpty(fetchCars)) {
      fetchCars();
    }
  }, []);

  let displayCarDetail;
  if(!isEmpty(fetchedCars)){
    let oneCar = fetchedCars[props.brand]['model'][props.model];
    displayCarDetail = (
      <div className={styles.container}>
        <img className={styles.image} src={oneCar.image} alt="it's a car detail data"/>
        <h4 className={styles.prices}>Price: {oneCar.price} </h4>
        <p className={styles.descripe}>design: {oneCar.introduction} </p>
        <Link to={"/cars/"+props.brand+"/"+props.model+"/ordersDetail"} >
          <button className={styles.button} type="button">Order it</button>
        </Link>
      </div>
    );
  }else{
    displayCarDetail=null;
  }

  return (
    displayCarDetail
  );
}

export default CarDetail;
