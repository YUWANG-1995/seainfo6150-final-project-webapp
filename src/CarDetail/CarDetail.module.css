.container{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.image{
  width: 800px;
  height: 500px;
}

.prices{
  font-size: 1.8rem;
}

.descripe{
  font-weight: bold;
  font-style: italic;
  font-size: 1.2em;
}

.button{
  background-color: #2f54eb;
  color: white;
  border: 2px solid #597ef7;
  border-radius: 5px;
  padding: 5px 10px;
  margin-top: 10px;
}
