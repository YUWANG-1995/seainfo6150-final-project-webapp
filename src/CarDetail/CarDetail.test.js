import React from "react";
import { render } from "@testing-library/react";
import CarDetail from "./CarDetail.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the CarDetail Component Correctly", () => {
  const test = {
    brand: 'BMW',
    model: {
      "X1": {
        modelId:"X1",
        price: "$35,400",
        image: "https://cache.bmwusa.com/cosy.arox?pov=walkaround&client=byo&vehicle=21XA&paint=P0300&brand=WBBM&sa=S01XV,S01XV,S0302,S0316,S03AG,S03AT,S03L8,S0420,S0508,S0775,S07HW&fabric=FKCSW&RESP=PNG&quality=90&width=800&sharpness=100&BKGND=TRANSPARENT&angle=60",
        introduction: "2.0-liter BMW TwinPower Turbo inline 4-cylinder, Front-wheel drive"
        }
    }
  }
  const { container } = render(
    <BrowserRouter>
      <CarDetail brand={test.brand} model={test.model} />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
