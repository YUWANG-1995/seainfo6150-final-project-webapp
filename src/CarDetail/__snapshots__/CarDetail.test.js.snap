// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CarDetail Component Correctly 1`] = `<div />`;
